Rewritten matrix component with matrix-nio sdk and with new features

szimszon opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This pull request is about to replace the old matrix component which uses [matrix-client sdk](https://pypi.org/project/matrix-client/) with a newer [matrix-nio sdk](https://matrix-nio.readthedocs.io/en/latest/index.html). It has the possibility to switch on end to end encryption in the feature. 

Right now the component has the same features as the old one and more.

New features are:

* send messages markdown formatted
* send messages as notice
* send local or remote image

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# The Matrix component
matrix:
  homeserver: https://matrix.org
  username: "@my_matrix_user:matrix.org"
  password: supersecurepassword
  rooms:
    - "#hasstest:matrix.org"
    - "#someothertest:matrix.org"
  commands:
    - word: testword
      name: testword
      rooms:
        - "#someothertest:matrix.org"
    - expression: "My name is (?P<name>.*)"
      name: introduction

notify:
  - name: matrix_notify
    platform: matrix
    default_room: "#hasstest:matrix.org"
    default_markdown: false
    default_notice: false

automation:
  - alias: 'React to !testword'
    trigger:
      platform: event
      event_type: matrix_command
      event_data:
        command: testword
    action:
      service: notify.matrix_notify
      data:
        message: "It looks like you wrote !testword"
  - alias: 'React to an introduction'
    trigger:
      platform: event
      event_type: matrix_command
      event_data:
        command: introduction
    action:
      service: notify.matrix_notify
      data:
        title: "Response"
        message: "Hello {{trigger.event.data.args['name']}}"
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

szimszon committed:
Rewritten matrix component with matrix-nio sdk and with new features

szimszon committed:
Fix most of the warnings...

szimszon committed:
Fix for C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)

szimszon committed:
Try to fix too many nested loops

szimszon committed:
Fix _command...

szimszon commented:
I wonder if somebody could help me.
The **Run tests Python 3.8 (group 1)** fails with this error message:

```
=========================== short test summary info ============================
ERROR tests/components/apns/test_notify.py
Error: Process completed with exit code 1.
```
It doesn't seem to be related to matrix. **apns** component test is failing.

What could I do? From the matrix component point of view is it still a problem?

homeassistant commented:
Hello @szimszon,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`6eaf5acb0b1a13a7b96edb093569c079f51dff1b`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;core&#x2F;pull&#x2F;46837&#x2F;commits&#x2F;6eaf5acb0b1a13a7b96edb093569c079f51dff1b) - This commit has something that looks like an email address (szimszon@oregpreshaz.eu). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

tkdrob committed:
Use core constants for dht (#46029)

jjlawren committed:
Fix Plex showing removed shared users (#46971)

milanmeu committed:
Fix typo in fireservicerota strings (#46973)

raman325 committed:
Add zwave_js.refresh_value service (#46944)

* add poll_value service

* switch vol.All to vol.Schema

* more relevant log message

* switch service name to refresh_value, add parameter to refresh all watched values, fix tests

* rename parameter and create task for polling command so we don't wait for a response

* raise ValueError for unknown entity

* better error message

* fix test

raman325 committed:
deep copy zwave_js state in test fixtures so tests are more isolated (#46976)

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Automatically create HomeKit accessory mode entries (#46473)

When we set up HomeKit, we asked users if they wanted
to create an entry in bridge or accessory mode.

This approach required the user to understand how HomeKit works and
choose which type to create.

When the user includes the media player or camera domains,
we exclude them from the bridge and create the additional entries
for each entity in accessory mode.

farmio committed:
Update xknx to 0.17.1 (#46974)

marcelveldt committed:
Extend zwave_js discovery scheme for lights (#46907)

natekspencer committed:
Add switches and sensors to Litter-Robot (#46942)

allenporter committed:
Make FAN_ON use the max duration rather than 15 min default (#46489)

StevenLooman committed:
Handle device IP change in upnp (#46859)

frenck committed:
Add template support to service targets (#46977)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

szimszon commented:
The error is about some import in apns test and not related to my PR:

```
==================================== ERRORS ====================================
__ ERROR collecting tests/components/apns/testnotify.py __
ImportError while importing test module '/w/core/core/tests/components/apns/test_notify.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.8/importlib/init__.py:127: in import_module...
```

What next?

farmio committed:
Validate KNX addresses (#46933)

szimszon commented:
**raman325** found out that matrix-nio use h2 in my matrix component and also apns is use it too. Still don't know how to resolve this best.

raman325 committed:
Add ClimaCell weather integration (#36547)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

szimszon commented:
```
venv/lib/python3.8/site-packages/h2/connection.py:13: in <module>
    from hyperframe.frame import (
E   ImportError: cannot import name 'ExtensionFrame' from 'hyperframe.frame' (/__w/core/core/venv/lib/python3.8/site-packages/hyperframe/frame.py)
```
I have a hyperframe installed in my dev environment with ExtensionFrame class available:
![image](https://user-images.githubusercontent.com/745542/108974038-54c72f80-7685-11eb-97b0-d2877641e488.png)

MartinHjelmare committed:
Save mysensors gateway type in config entry (#46981)

meichthys committed:
Mullvad VPN (#44189)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>

MartinHjelmare committed:
Improve mysensors config flow (#46984)

bramkragten committed:
Fix TTS services name (#46988)

frenck committed:
Mullvad integration improvements (#46987)

frenck committed:
Merge branch 'master' into dev

bieniu committed:
Catch AuthRequired exception in confirm discovery step for Shelly config flow (#46135)

Noltari committed:
Restore Tado binary sensor attributes (#46069)

adrian-vlad committed:
Add enable and disable services for recorder (#45778)

raman325 committed:
Migrate zwave_js entities to use new unique ID format (#46979)

* migrate zwave_js entities to use new unique ID format

* remove extra param from helper

* add comment to remove migration logic in the future

* update comment

* use instance attribute instead of calling functino on every state update

MartinHjelmare committed:
Fix zwave_js config flow server version timeout (#46990)

fabaff committed:
Set awesomeversion to 21.2.3 (#46989)

ludeeus committed:
Add addon selector (#46789)

raman325 committed:
Climacell fixes: Use common keys for strings, fix temp_low measurement, add windy condition (#46991)

* use common keys for lat and long

* additional fixes

* Update homeassistant/components/climacell/strings.json

Co-authored-by: Milan Meulemans <milan.meulemans@live.be>

Co-authored-by: Milan Meulemans <milan.meulemans@live.be>

milanmeu committed:
Use location common key reference in totalconnect (#46995)

cgtobi committed:
Add Netatmo device trigger (#45387)

Co-authored-by: J. Nick Koston <nick@koston.org>

milanmeu committed:
Remove recursive key reference (#46999)

marcelveldt committed:
Add number platform to Z-Wave JS (#46956)

* add number platform to zwave_js integration
* add discovery scheme for thermostat valve control, using number platform

Co-authored-by: kpine <keith.pine@gmail.com>

raman325 committed:
Setup config entry even if vizio device is unreachable (#46864)

emontnemery committed:
Support value_template in MQTT triggers (#46891)

* Support value_template in MQTT triggers

* Rename value_template to payload_template

* Revert "Rename value_template to payload_template"

This reverts commit 902094eefc6612e6b5c3bdb7440520af050c7f20.

cgtobi committed:
Add Sonos media browser image proxy (#46902)

jjlawren committed:
Use Plex server URL as config entry title (#47010)

jjlawren committed:
Bump plexapi to 4.4.0 (#47007)

ntilley905 committed:
Add FAA Delays Integration (#41347)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

chemelli74 committed:
Add missing tilt icon to Shelly tilt sensor (#46993)

bramkragten committed:
Update frontend to 20210224.0 (#47013)

marcelveldt committed:
Change Z-Wave JS discovery logic to adopt changes to DeviceClass (#46983)

Co-authored-by: raman325 <7243222+raman325@users.noreply.github.com>

frenck committed:
Bump version to 2021.4.0dev0 (#47017)

balloob committed:
Convert discovery helper to use dispatcher (#47008)

homeassistant committed:
[ci skip] Translation update

fabaff committed:
Upgrade TwitterAPI to 2.6.8 (#47019)

starkillerOG committed:
Add Xiaomi Miio fan config flow (#46866)

* Miio fan config flow

* fix styling and imports

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* rename device -> entity

* fix indent

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Fix zwave_js unique ID migration logic (#47031)

bdraco committed:
Ensure doorbird events are re-registered when changing options (#46860)

- Fixed the update listener not being unsubscribed

- DRY up some of the code

- Fix sync code being called in async

- Reduce executor jumps

balloob committed:
Remove deprecated credstash + keyring (#47033)

FHeilmann committed:
hm climate: Return PRESET_NONE instead of None (#47003)

Signed-off-by: Florian Heilmann <Florian.Heilmann@gmx.net>

chemelli74 committed:
Fix missing Shelly external input (#47028)

* Add support for external input (Shelly 1/1pm add-on)

* Make external sensor naming consistent

* Fix case consistency

Kane610 committed:
Normally there should only be one battery sensor per device from deCONZ. (#46761)

With these Danfoss devices each endpoint can report its own battery state.

fabaff committed:
Upgrade icmplib to 2.0.2 (#47039)

balloob committed:
Clean up discovery integration (#47022)

* Clean up discovery integration

* Fix tests

* Remove discovery step from freebox

fabaff committed:
Upgrade pyowm to 3.2.0 (#47042)

fabaff committed:
Upgrade sendgrid to 6.6.0 (#47041)

joogps committed:
Add remote control platform to Panasonic Viera (#42450)

* Adding remote platform

* Update homeassistant/components/panasonic_viera/remote.py

Simplify entity creation

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Use Pytest fixture

* Use Pytest fixtures and assert service calls

* Adding conftest.py and organizing tests

* Reorganizing tests

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bieniu committed:
Bump gios library to version 0.2.0 (#47050)

Melantrix committed:
Replace wrong domain returned from xbox api 2.0 (#47021)

* Change solution to use yarl lib

* Add check if base url needs changing

* Actively remove padding query instead of omitting

* Fixed popping the wrong query

* Change explaination about removing mode query

Quentame committed:
Add zeroconf discovery to Freebox (#47045)

* Add zeroconf discovery to Freebox

- remove deprecated discovery
- tried with SSDP too but the presentation URL is not the same (*.fbxos.fr for zeroconf, http://mafreebox.freebox.fr/ for SSDP)
- so config entry unique_id should be the MAC (included into SSDP, but not zeroconf, can be retrieve from `fbx.system.get_config()`)
- DHCP discovery might be added in the future too

* host and port are required on zeroconf

* cleanup in other PR

cgtobi committed:
Clean up Netatmo webhook handler (#47037)

cyberjunky committed:
Bump python-garminconnect to 0.1.19 to fix broken api (#47020)

bdraco committed:
Fix bond typing in config flow (#47055)

ludeeus committed:
Use dispatch instead of eventbus for supervisor events (#46986)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bramkragten committed:
Updated frontend to 20210225.0 (#47059)

bramkragten committed:
Revert CORS changes for my home assistant (#47064)

* Revert CORS changes for my home assistant

* Update test_init.py

* Update test_init.py

homeassistant committed:
[ci skip] Translation update

raman325 committed:
catch ValueError when unique ID update fails because its taken and remove the duplicate entity (#47072)

cgarwood committed:
Fix Z-Wave JS API docstrings (#47061)

bdraco committed:
Ensure hue options show the defaults when the config options have not yet been saved (#47067)

balloob committed:
Bump Z-Wave JS Server Python to 0.20.0 (#47076)

balloob committed:
Upgrade aiohttp to 3.7.4 (#47077)

chpego committed:
Upgrade youtube_dl to version 2021.02.22 (#47078)

* Upgrade youtube_dl to version 2021.02.22

* Update requirements_all.txt

emontnemery committed:
Bump pychromecast to 8.1.2 (#47085)

Commit:
Add support for Shelly SHBTN-2 device triggers (#46644)

emontnemery committed:
Bump pychromecast to 9.0.0 (#47086)

* Adapt to Pychromecast 9.0.0

* Bump pychromecast to 9.0.0

* Fix lint issues

raman325 committed:
Remove flaky climacell test (#47080)

rikroe committed:
Bump bimmer_connected to 0.7.15 and fix bugs (#47066)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

Shutgun committed:
Add suggested_area support to devolo Home Control (#47063)

agners committed:
Add new machine generic-x86-64 to build matrix (#47095)

The Intel NUC machine runs on most UEFI capable x86-64 machines today.
Lets start a new machine generic-x86-64 which will replace intel-nuc
over time.

cgarwood committed:
Guard zwave_js missing nodes in websocket api (#47096)

bdraco committed:
Add suggested area to hue (#47056)

elupus committed:
Add support for v6 features to philips js integration (#46422)

chemelli74 committed:
Bump aioshelly to 0.6.1 (#47088)

marcelveldt committed:
Fix Z-Wave JS discovery schema for thermostat devices (#47087)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

allenporter committed:
Bump google-nest-sdm to v0.2.12 to improve API call error messages (#47108)

balloob committed:
Use async_capture_events to avoid running in executor (#47111)

thecode committed:
Fix Shelly RGBW (#47116)

homeassistant committed:
[ci skip] Translation update

bramkragten committed:
Update frontend to 20210226.0 (#47123)

bdraco committed:
Provide a human readable exception for the percentage util (#47121)

bdraco committed:
Handle lutron_caseta fan speed being none (#47120)

allenporter committed:
Improve handling for recording start of nest cam stream (#47144)

* Improve handling for start of nest cam stream

Add negative_cts_offsets to segment container options in order to better handle recording at the start of a stream. Nest streams start off with a negative offset, and if the segment container does not support it, then it adjusts the timestamps making it out of order with the next segment as described in issue #46968

* Update homeassistant/components/stream/__init__.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Kane610 committed:
Change device class of window covers to shade (#47129)

homeassistant committed:
[ci skip] Translation update

bieniu committed:
Bump airly library to version 1.1.0 (#47163)

tkdrob committed:
Clean up mqtt_room (#46882)

agners committed:
Bump builder to get generic-x86-64 nightly builds (#47164)

balloob committed:
Fix the updater schema (#47128)

dmulcahey committed:
Bump ZHA quirks to 0.0.54 (#47172)

emontnemery committed:
Tweak Tasmota fan typing (#47175)

mib1185 committed:
Apply recommendations to synology_dsm (#47178)

allenporter committed:
Fix flaky hls keepalive test (#47186)

Remove a call to stream.start() which is issued before the test is fully
setup (e.g. keepalive is not set to True, and mock calls are not registered)

bdraco committed:
Update HAP-python to 3.3.1 (#47180)

Fixes disconnect when setting a single char fails
https://github.com/ikalchev/HAP-python/compare/v3.3.0...v3.3.1

emontnemery committed:
Fix MQTT trigger where wanted payload may be parsed as an integer (#47162)

emontnemery committed:
Update MQTT device triggers with support for templates (#47142)

starkillerOG committed:
Fix Xiaomi Miio discovery (#47134)

ajschmidt8 committed:
Update AlarmDecoder dependency (#46841)

mdz committed:
Remove turn_on and turn_off from SmartTub pump switches (#47184)

mdz committed:
Move SmartTub climate constants to module level (#47190)

esev committed:
Explain why should_pool is True initially for wemo (#47191)

mdz committed:
Cleanup SmartTub filtration cycles (#47192)

homeassistant committed:
[ci skip] Translation update

allenporter committed:
Handle stream failures in recorder (#47151)

* Handle stream failures in recorder

Fail gracefully with an error message when the recorder is invoked with no segments due to a stream failure.

* Update homeassistant/components/stream/recorder.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

bdraco committed:
Enforce typing in bond (#47187)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bachya committed:
Bump simplisafe-python to 9.6.7 (#47206)

bdraco committed:
Increment the homekit config version when restarting (#47209)

If an entity changes between restart the iOS/controller
device may have cached the old chars for the accessory.
To force the iOS/controller to reload the chars, we
increment the config version when Home Assistant restarts

bdraco committed:
Bump HAP-python to 3.3.2 to fix unavailable condition on restart (#47213)

Fixes

https://github.com/ikalchev/HAP-python/compare/v3.3.1...v3.3.2

cdce8p committed:
Update pylint (#47205)

agners committed:
Fix generic-x86-64 build (#47214)

Replace the wrong Docker Hub repository slipped in during testing.

raman325 committed:
Add hassio addon_update service and hassio config entry with addon and OS devices and entities (#46342)

* add addon_update service, use config flow to set up config entry, create disabled sensors

* move most of entity logic to common entity class, improve device info, get rid of config_flow user step

* fix setup logic

* additional refactor

* fix refactored logic

* fix config flow tests

* add test for addon_update service and get_addons_info

* add entry setup and unload test and fix update coordinator

* handle if entry setup calls unload

* return nothing for coordinator if entry is being reloaded because coordinator will get recreated anyway

* remove entry when HA instance is no longer hassio and add corresponding test

* handle adding and removing device registry entries

* better config entry reload logic

* fix comment

* bugfix

* fix flake error

* switch pass to return

* use repository attribute for model and fallback to url

* use custom 'system' source since hassio source is misleading

* Update homeassistant/components/hassio/entity.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* update remove addons function name

* Update homeassistant/components/hassio/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* fix import

* pop coordinator after unload

* additional fixes

* always pass in sensor name when creating entity

* prefix one more function with async and fix tests

* use supervisor info for addons since list is already filtered on what's installed

* remove unused service

* update sensor names

* remove added handler function

* use walrus

* add OS device and sensors

* fix

* re-add addon_update service schema

* add more test coverage and exclude entities from tests

* check if instance is using hass OS in order to create OS entities

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cdce8p committed:
Improve CI workflow (#46696)

cgtobi committed:
Revert const replacement in fritzbox_callmonitor (#47211)

bdraco committed:
Remove griddy integration (#47218)

maxwroc committed:
Fix number of reported issues by github integration (#47203)

farmio committed:
KNX services send and event_register accept multiple group addresses (#46908)

* send and event_register service accept lists of group addresses

* remove lambda

* object selector for lists

* knx.read takes lists too

farmio committed:
Deprecate knx config_file (#46874)

* deprecate config_file

* removed cv.deprecated for now, added persistent notification

* Update homeassistant/components/knx/__init__.py

Co-authored-by: Philip Allgaier <philip.allgaier@gmx.de>

* remove notification, add cv.deprecated again

* Update homeassistant/components/knx/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* remove cv.deprecated again

Co-authored-by: Philip Allgaier <philip.allgaier@gmx.de>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

emontnemery committed:
Fix race when disabling config entries (#47210)

* Fix race when disabling config entries

* Remove unused constant

starkillerOG committed:
Fix Xiaomi Miio flow unique_id for non discovery flows (#47222)

frenck committed:
Restore pylint concurrency (#47221)

mdz committed:
Parameterize SmartTub tests (#47189)

* Parameterize SmartTub tests

* parameterize light service calls

* remove stray print()

* add comment

frenck committed:
Upgrade coverage to 5.5 (#47227)

frenck committed:
Upgrade spotipy to 2.17.1 (#47228)

jdeath committed:
Bump mcstatus to 5.1.1 (#47169)

dermotduffy committed:
Overhaul command_line tests (#46682)

thecode committed:
Fix Shelly Polling (#47224)

dermotduffy committed:
Minor Hyperion mypy cleanups (#45765)

raman325 committed:
Address late hassio review (#47229)

* hassio code cleanup to address comments in #46342

* fix code

farmio committed:
KNX address constant (#47196)

emontnemery committed:
Pass variables to initial evaluation of template trigger (#47236)

* Pass variables to initial evaluation of template trigger

* Add test

* Clarify test

emontnemery committed:
Add support for a list of known hosts to Google Cast (#47232)

bdraco committed:
Fix lutron caseta fan handling of speed off (#47244)

elupus committed:
Add remote control support to philips_js (#47249)

raman325 committed:
Add suggested area for zwave_js devices (#47250)

bramkragten committed:
Update frontend to 20210301.0 (#47252)

bachya committed:
Bump simplisafe-python to 9.6.8 (#47241)

balloob committed:
Revert "Fix the updater schema (#47128)" (#47254)

This reverts commit 98be703d90e44efe43b1a17c7e5243e5097b00b1.

frenck committed:
Fix duplicate template handling in Persistent Notifications (#47217)

frenck committed:
Deprecate LIFX Legacy integration (#47235)

milanmeu committed:
Fix typo in plaato strings (#47245)

bdraco committed:
Fix harmony failing to switch activities when a switch is in progress (#47212)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

stephan192 committed:
Remove rounding from The Things Network (#47157)

marcelveldt committed:
Update color logic for zwave_js light platform (#47110)

Co-authored-by: Raman Gupta <7243222+raman325@users.noreply.github.com>

emontnemery committed:
Add battery sensor for gogogate2 wireless door sensor (#47145)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Fix typing on fan percentage (#47259)

scop committed:
Lint suppression cleanups (#47248)

* Unused pylint suppression cleanups

* Remove outdated pylint bug references

* Add flake8-noqa config and note to run it every now and then

* Add codes to noqa's

* Unused noqa cleanups

tkdrob committed:
Clean up constants (#46948)

* Clean up constants

* clean up humidifier constants

* fix tests

* fix prometheus tests

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

Quentame committed:
Add init test to Freebox (#46998)

* Add init test to Freebox

* Review : more readable conftest

* Expect 2 blank lines between defs

* Review : Not I/O in the event loop

* Fix test_setup test

* remove useless const

* Review : mock setup methods

* Add service test

* Add import test

Nick-Adams-AU committed:
Add services for izone airflow min/max (#45727)

* Create airflow_min and airflow_max services for the izone component
* Bump pizone library requirement

emontnemery committed:
Make MQTT number respect retain setting (#47270)

balloob committed:
Fix Alexa doorbells (#47257)

frenck committed:
Fix izone flake8 error (#47276)

rlehfeld committed:
Add force_update to tasmota sensors (#47052)

* Add force update also to non-binary sensors as e.g. POWER Measurement agerage cannot be calculated otherwise. This is the same behavior as set with the obsolete tasmota detection

* add tests in binary_sensor and test_sensor for force_update flag

* satisfy flake8

* next try for force_update test but this time on the entity object which is the correct level

* once again satisfy flake8

* one more try for a test

* fix typo

* satisfy black

bramkragten committed:
Update frontend to 20210302.0 (#47278)

bachya committed:
Bump simplisafe-python to 9.6.9 (#47273)

Adminiuga committed:
Update ZHA dependencies (#47282)

scop committed:
Upgrade isort to 5.7.0 (#47279)

https://pycqa.github.io/isort/CHANGELOG/#570-december-30th-2020

Quentame committed:
Add disk sensor to Freebox (#46689)

* Add disk sensor to Freebox

* Add debug logging into sensors

* Remove useless sensor[X] assignement in disk sensor

Quentame committed:
Uniformize platform setup (#47101)

* A platform is not a component

* Fix dynalite

* SUPPORTED_PLATFORMS --> PLATFORMS

* In tests

* In tests 2

* Fix SmartThings

* Fix ZHA test

* Fix Z-Wave

* Revert Z-Wave

* Use PLATFORMS const in ambient_station

* Fix ihc comment

chemelli74 committed:
Limit log spam by ConfigEntryNotReady (#47201)

balloob committed:
Clean up secret loading (#47034)

frenck committed:
Upgrade pillow to 8.1.1 (#47223)

cgtobi committed:
Fix issue when setting boost preset for a turned off Netatmo thermostat (#47275)

raman325 committed:
Add raw values to zwave_js value notification event (#47258)

* add value_raw to value notification event that always shows the untranslated state value

* add property key and property to event params

raman325 committed:
Convert climacell forecast timestamp to isoformat so that UI shows the right times (#47286)

raman325 committed:
Bump zwave-js-server-python to 0.20.1 (#47289)

MartinHjelmare committed:
Add zwave_js add-on manager (#47251)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

farmio committed:
KNX remove custom deprecation warnings (#47238)

raman325 committed:
Correct climacell device info (#47292)

bdraco committed:
Add activity properties to remote entity model (#47237)

mdz committed:
bump python-smarttub to 0.0.19 (#47294)

milanmeu committed:
Remove name from keenetic-ndms2 strings (#47113)

emontnemery committed:
Correct gogogate2 battery sensor attributes (#47302)

Nick-Adams-AU committed:
Update izone services.yaml and remove entity_id from schema. (#47305)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bramkragten committed:
Update frontend to 20210302.3 (#47310)

MartinHjelmare committed:
Fix Supervisor platform coordinator data lookup (#47308)

elupus committed:
Philips JS correct post review comments (#47247)

emontnemery committed:
Improve behaviour when disabling or enabling config entries (#47301)

balloob committed:
Simplify switch light (#47317)

tkdrob committed:
Fix grammar in pi_hole logs (#47324)

firstof9 committed:
Add LZW36 device schema to zwave_js discovery (#47314)

* Add LZW26 device schema to discovery

Co-authored-by: @kpine

* Update homeassistant/components/zwave_js/discovery.py

Co-authored-by: kpine <keith.pine@gmail.com>

* Add tests

* Fix test

Co-authored-by: kpine <keith.pine@gmail.com>

emontnemery committed:
Initial automation tracing (#46755)

* Initial prototype of automation tracing

* Small fixes

* Lint

* Move trace helpers to its own file

* Improve trace for state and numeric_state conditions

* Tweaks + apply suggestions from code review

* Index traces by automation_id, trace while script is running

* Refactor condition tracing

* Improve WS API to get traces for single automation

* Add tests

* Fix imports

* Fix imports

* Address review comments

* Cap logging of loops

* Remove unused ContextVar action_config

frenck committed:
Fix secrets in files included via include_dir_list (#47350)

amelchio committed:
Catch ConditionError in generic_thermostat climate (#47359)

starkillerOG committed:
Fix Xiaomi Miio setup of switch entity for lumi.acpartner.v3 (#47345)

bdraco committed:
Map silent as a preset mode for fan backcompat (#47396)

The original change did not map silent as a preset mode
because it was not clear if it was a speed or a preset.

esev committed:
Revert "Speed-up wemo discovery (#46821)" (#47392)

This reverts commit 6e52b26c06098052d379065b00f570c2a44653e1.

frenck committed:
Don't raise on known non-matching states in numeric state condition (#47378)

bramkragten committed:
Update frontend to 20210302.4 (#47383)

MartinHjelmare committed:
Fix zwave_js manual reconfiguration of add-on managed entry (#47364)

MartinHjelmare committed:
Fix access of missing zwave_js climate unit value (#47380)

syssi committed:
Fix measurement unit (Closes: #47390) (#47398)

raman325 committed:
Don't convert Climacell forecast temperatures to celsius because platform does it automatically (#47406)

bachya committed:
Fix AirVisual exception when config entry contains old integration type (#47405)

raman325 committed:
Fix Climacell timezone issue with daily forecasts (#47402)

frenck committed:
Fix older Roborock models (#47412)

ppetru committed:
Update Solax library to 0.2.6 (#47384)

MartinHjelmare committed:
Make zwave_js add-on manager more flexible (#47356)

cgtobi committed:
Update browse_media.py (#47414)

Cooper-Dale committed:
Fix typo in docs link for forked_daapd (#47413)

corrected link to existing site

cpainchaud committed:
Fix RFLink TCP KeepAlive error log (#47395)

raman325 committed:
Bump zwave-js-server-python to 0.21.0 (#47408)

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

MartinHjelmare committed:
Only create snapshot if add-on update will be done (#47424)

systemcrash committed:
Spellcheck on Synology component (#47451)

uvjustin committed:
Add allenporter to stream codeowners (#47431)

amelchio committed:
Limit log spam by ESPHome (#47456)

raman325 committed:
Bump zwave-js-server-python to 0.21.1 (#47464)

bramkragten committed:
Update frontend to 20210302.5 (#47462)

elupus committed:
Bump version with fix for v1 (#47458)

frenck committed:
Fix Hue scene overriding Hue default transition times (#47454)

tkdrob committed:
Clean up constants (#47323)

emlove committed:
Prevent Zerproc leaving open unnecessary connections (#47401)

* Zerproc: Prevent leaving open unnecessary connections

* Fix config entry unloading

functionpointer committed:
Use conn_made callback in MySensors (#47463)

mvn23 committed:
Update pyotgw to 1.1b1 (#47446)

frenck committed:
Deprecate HomeKit auto start (#47470)

raman325 committed:
Update zwave_js.refresh_value service description (#47469)

emlove committed:
Convert kulersky to use new async backend (#47403)

Kane610 committed:
Increase test coverage of UniFi integration (#46347)

* Increase coverage of init

* Increase coverage of config_flow

* Improve coverage of controller

* Minor improvement to switch test

* Fix review comment

* Mock websocket class

* Replace the rest of the old websocket event tests

* Improve websocket fixture for cleaner tests

* Fix typing

* Improve connection state signalling based on Martins feedback

* Improve tests of reconnection_mechanisms based on Martins review comments

* Fix unload entry

* Fix isort issue after rebase

* Fix martins comment on not using caplog

* Fix wireless clients test

* Fix martins comments on wireless clients test

cgtobi committed:
Fix issue at Netatmo startup (#47452)

frenck committed:
Typing tweak to the Elgato integration (#47471)

Kane610 committed:
Improve restoring UniFi POE entity state (#47148)

* Improve restoring data and better handling when the restore data is empty
Improve readability of some logic related to POE clients

* There is no need to check clients_all in Switch platform

* Add better tests when restoring state

* Port except handling shouldn't be needed anymore

* Walrusify get_last_state

Adminiuga committed:
Update ZHA dependencies (#47479)

ollo69 committed:
Move AsusWrt sensors update logic in router module (#46606)

pnbruckner committed:
Bump amcrest package version to 1.7.1 (#47483)

frenck committed:
Complete typing on TwenteMilieu integration (#47480)

balloob committed:
Raise error instead of crashing when template passed to call service target (#47467)

frenck committed:
Complete typing on Verisure integration (#47482)

amelchio committed:
Fix Sonos polling mode (#47498)

FidgetyRat committed:
Add OPENING & CLOSING state to MySensors cover (#47285)

* Added OPENING & CLOSING State Support

Added support for OPENING and CLOSING states using a combination of
the required V_ variables. Simplified the determination of the
cover's state by use of a new enumeration and single method allowing
the state to be used by all three HomeAssistant query methods.

* Fixes for HomeAssistant Style

Corrections to style to allow flake8, isort, and black to pass.

* Peer Review Changes

Added @unique to the main enumeration. Removed unnecessary parens from
door state logic. Reordered CLOSING and CLOSED in the enumeration.

scop committed:
Upgrade upcloud-api to 1.0.1 (#47501)

https://github.com/UpCloudLtd/upcloud-python-api/releases/tag/0.4.6
https://github.com/UpCloudLtd/upcloud-python-api/releases/tag/v1.0.0
https://github.com/UpCloudLtd/upcloud-python-api/releases/tag/v1.0.1

emontnemery committed:
Merge action and condition traces (#47373)

* Merge action and condition traces

* Update __init__.py

* Add typing to AutomationTrace

* Make trace_get prepare a new trace by default

* Correct typing of trace_cv

* Fix tests

uvjustin committed:
Disable audio stream when ADTS AAC detected (#47441)

* Disable audio stream when ADTS AAC detected

* Use context manager for memoryview

* Fix tests

* Add test

* Fix tests

* Change FakePacket bytearray size to 3

natekspencer committed:
Adjust litterrobot tests and code to match guidelines (#47060)

* Use SwitchEntity instead of ToggleEntity and adjust test patches as recommended

* Move async_create_entry out of try block in config_flow

* Patch pypi package instead of HA code

* Bump pylitterbot to 2021.2.6, fix tests, and implement other code review suggestions

* Bump pylitterbot to 2021.2.8, remove sleep mode start/end time from vacuum, adjust and add sensors for sleep mode start/end time

* Move icon helper back to Litter-Robot component and isoformat times on time sensors

szimszon committed:
Merge branch 'matrix-nio' of https://github.com/szimszon/core into matrix-nio

szimszon commented:
I rebased my modul with up-to-date dev branch from homeassistant/core.

Still got error:
```
ERROR tests/components/apns/test_notify.py
Error: Process completed with exit code 1.
```
It is still unrelated to my modul.

elupus commented:
Your rebase failed. You managed to get a large amount of unrelated commits into your branch.

szimszon commented:
I created a clean new pull request: https://github.com/home-assistant/core/pull/47529 I hope it will fit.
